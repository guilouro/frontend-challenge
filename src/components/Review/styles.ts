import styled, { css } from 'styled-components'

type StarProps = {
  isGold: boolean
}

export const Star = styled.svg<StarProps>`
  ${({ isGold }) => css`
    color: ${isGold ? '#f1c40f' : '#e7e7e9'};
  `}
`
