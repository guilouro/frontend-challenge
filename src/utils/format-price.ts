export default function formatPrice(price: number | bigint): string {
  return new Intl.NumberFormat('de-DE', {
    style: 'currency',
    currency: 'EUR'
  }).format(price)
}
