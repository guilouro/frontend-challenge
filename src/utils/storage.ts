import { CartObject } from 'hooks/use-cart'

const APP_KEY = 'BARKYN_CART'

export function getStorageItem() {
  if (typeof window === 'undefined') return

  const data = window.localStorage.getItem(APP_KEY)
  return JSON.parse(data || '[]')
}

export function setStorageItem(value: CartObject) {
  if (typeof window === 'undefined') return

  const data = JSON.stringify(value)
  return window.localStorage.setItem(APP_KEY, data)
}
