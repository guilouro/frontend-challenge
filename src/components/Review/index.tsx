import * as S from './styles'

export type ReviewProps = {
  review?: number
}

const Review = ({ review = 0 }: ReviewProps) => (
  <div role="img" aria-label={`${review} stars of 5`}>
    {[...Array(5).keys()].map((index) => (
      <S.Star
        isGold={review >= index + 1}
        key={index}
        height="16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 24 24"
      >
        <g fill="none" fillRule="evenodd">
          <path d="m0 0h24v24h-24z" />
          <path
            d="m12 18-5.26 2.765a1 1 0 0 1 -1.45-1.054l1.004-5.857-4.256-4.148a1 1 0 0 1 .554-1.706l5.881-.854 2.63-5.329a1 1 0 0 1 1.794 0l2.63 5.329 5.88.854a1 1 0 0 1 .555 1.706l-4.256 4.148 1.005 5.857a1 1 0 0 1 -1.451 1.054z"
            fill="currentColor"
          />
        </g>
      </S.Star>
    ))}
  </div>
)

export default Review
