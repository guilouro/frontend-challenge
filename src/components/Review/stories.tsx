import { Story, Meta } from '@storybook/react'
import Review, { ReviewProps } from '.'

export default {
  title: 'Review',
  component: Review,
  args: {
    review: 2
  },
  argTypes: {
    review: {
      control: { type: 'range', min: 0, max: 5 }
    }
  }
} as Meta

export const Default: Story<ReviewProps> = (args) => <Review {...args} />
