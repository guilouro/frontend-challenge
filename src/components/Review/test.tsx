import { render, screen } from 'utils/test-utils'

import Review from '.'

describe('<Review />', () => {
  it('should render correctly without stars', () => {
    const { container } = render(<Review />)
    expect(
      screen.getByRole('img', { name: /0 stars of 5/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render with 3 stars', () => {
    const stars = 3
    render(<Review review={stars} />)

    const el = screen.getByRole('img', { name: /3 stars of 5/i })
    expect(el).toBeInTheDocument()

    el.querySelectorAll('svg').forEach((item, idx) => {
      expect(item).toHaveStyleRule(
        'color',
        idx + 1 <= stars ? '#f1c40f' : '#e7e7e9'
      )
    })
  })
})
