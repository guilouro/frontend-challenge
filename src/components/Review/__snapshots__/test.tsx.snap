// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Review /> should render correctly without stars 1`] = `
.c0 {
  color: #e7e7e9;
}

<div
  aria-label="0 stars of 5"
  role="img"
>
  <svg
    class="c0"
    height="16"
    viewBox="0 0 24 24"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      fill="none"
      fill-rule="evenodd"
    >
      <path
        d="m0 0h24v24h-24z"
      />
      <path
        d="m12 18-5.26 2.765a1 1 0 0 1 -1.45-1.054l1.004-5.857-4.256-4.148a1 1 0 0 1 .554-1.706l5.881-.854 2.63-5.329a1 1 0 0 1 1.794 0l2.63 5.329 5.88.854a1 1 0 0 1 .555 1.706l-4.256 4.148 1.005 5.857a1 1 0 0 1 -1.451 1.054z"
        fill="currentColor"
      />
    </g>
  </svg>
  <svg
    class="c0"
    height="16"
    viewBox="0 0 24 24"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      fill="none"
      fill-rule="evenodd"
    >
      <path
        d="m0 0h24v24h-24z"
      />
      <path
        d="m12 18-5.26 2.765a1 1 0 0 1 -1.45-1.054l1.004-5.857-4.256-4.148a1 1 0 0 1 .554-1.706l5.881-.854 2.63-5.329a1 1 0 0 1 1.794 0l2.63 5.329 5.88.854a1 1 0 0 1 .555 1.706l-4.256 4.148 1.005 5.857a1 1 0 0 1 -1.451 1.054z"
        fill="currentColor"
      />
    </g>
  </svg>
  <svg
    class="c0"
    height="16"
    viewBox="0 0 24 24"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      fill="none"
      fill-rule="evenodd"
    >
      <path
        d="m0 0h24v24h-24z"
      />
      <path
        d="m12 18-5.26 2.765a1 1 0 0 1 -1.45-1.054l1.004-5.857-4.256-4.148a1 1 0 0 1 .554-1.706l5.881-.854 2.63-5.329a1 1 0 0 1 1.794 0l2.63 5.329 5.88.854a1 1 0 0 1 .555 1.706l-4.256 4.148 1.005 5.857a1 1 0 0 1 -1.451 1.054z"
        fill="currentColor"
      />
    </g>
  </svg>
  <svg
    class="c0"
    height="16"
    viewBox="0 0 24 24"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      fill="none"
      fill-rule="evenodd"
    >
      <path
        d="m0 0h24v24h-24z"
      />
      <path
        d="m12 18-5.26 2.765a1 1 0 0 1 -1.45-1.054l1.004-5.857-4.256-4.148a1 1 0 0 1 .554-1.706l5.881-.854 2.63-5.329a1 1 0 0 1 1.794 0l2.63 5.329 5.88.854a1 1 0 0 1 .555 1.706l-4.256 4.148 1.005 5.857a1 1 0 0 1 -1.451 1.054z"
        fill="currentColor"
      />
    </g>
  </svg>
  <svg
    class="c0"
    height="16"
    viewBox="0 0 24 24"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      fill="none"
      fill-rule="evenodd"
    >
      <path
        d="m0 0h24v24h-24z"
      />
      <path
        d="m12 18-5.26 2.765a1 1 0 0 1 -1.45-1.054l1.004-5.857-4.256-4.148a1 1 0 0 1 .554-1.706l5.881-.854 2.63-5.329a1 1 0 0 1 1.794 0l2.63 5.329 5.88.854a1 1 0 0 1 .555 1.706l-4.256 4.148 1.005 5.857a1 1 0 0 1 -1.451 1.054z"
        fill="currentColor"
      />
    </g>
  </svg>
</div>
`
